### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by lomeroe**
Adding windows support to refactored kitchen-salt code.

### CODE REVIEW

**Comment by gtmanfred** [comment]
@lomeroe is it ok with you if we get #50 in first, and then if there are any more changes that need to be made, we can make them after that?

Thanks,
Daniel

**Comment by lomeroe** [comment]
I will take a look at it, been quite a while since I created it...I can't remember why I did my own PR instead of using #50 

I'm excited to see that saltstack has taken kitchen-salt on.

**Comment by gtmanfred** [comment]
I am excited too, though i have been really struggling and need to brush up on my ruby 😸

**Comment by lomeroe** [comment]
@gtmanfred I poked through this and it looks like the differences are:

1) I don't hard code c:\tmp as the temp directory, I use $env:Temp
2) I use a erb template for the windows minion install
3) salt-call command's retcode is returned/tested
4) I didn't add any tests

Not sure why I didn't just take that PR and make updates to it, looks like I "borrowed" some that PR's code.  Likely b/c of my weak git fu.

**Comment by barkingfoodog** [comment]
1. This makes plenty of sense, so I'm all for it
2. I've worked on re-basing #50 last night and noticed the shift to using `.erb` files. It makes sense to move the main installation script out to a separate file, so +1 for this item as well
3. Good deal. This was a TODO item I'd left in #50 
4. These would be pretty easy to port over

In short, I think we need to standardize on one of these PRs, since #74 doesn't really build on #50 in terms of the commit history. I'm cool to go with this one and can port over the tests if you'd like. They most make sure the scripts quack like a duck on unix and honk like a goose on Windows. ;)

I also noticed you're using the official `salt_bootstrap.ps1` script instead of chocolately, which is also just fine by me.

**Comment by gtmanfred** [comment]
Yeah, i like the ps1 script instead of chocolately too.

So if you could rebase this @lomeroe and the @barkingfooddog if you could port over the tests once we get this merged, i would appreciate it!

I am going to close #50 then.

Thanks

**Comment by barkingfoodog** [comment]
@gtmanfred will do. I'll watch for this to be merged and make a separate PR with the tests.

**Comment by lomeroe** [comment]
I still need to test this after the rebase, but wanted to put it up here so others could test as well...

**Comment by lomeroe** [comment]
@gtmanfred this tested out on my end...

File: Gemfile [hunk 1]
```diff
@@ -18,4 +18,4 @@ group :docker do
   gem 'kitchen-docker', '~> 2.1.0'
 end
 
-# vi: set ft=ruby :
+# vi: set ft=ruby :
```
Comment (gtmanfred) [inline Gemfile]: Can you fix this and add the newline at the end of the file back?

File: lib/kitchen/provisioner/salt_solo.rb [hunk 2]
```diff
@@ -134,27 +148,42 @@ def create_sandbox
 
       def init_command
         debug("Initialising Driver #{name} by cleaning #{config[:root_path]}")
-        cmd = "mkdir -p '#{config[:root_path]}';"
-        cmd += <<-INSTALL
-          #{config[:init_environment]}
-        INSTALL
+        if windows_os?
+          cmd = "rm ""#{config[:root_path]}"" -Recurse -Force; mkdir -Path ""#{config[:root_path]}"
```
Comment (gtmanfred) [inline lib/kitchen/provisioner/salt_solo.rb]: Does this really need to be recursively removed first?

**Review by gtmanfred — CHANGES_REQUESTED** [review]
Just one nitpicky thing and a question about another, and then it should be good to merge.

**Comment by gtmanfred** [comment]
Would you also double check that the changes made here also work on windows?
https://github.com/saltstack/kitchen-salt/pull/72/files

Thanks
Daniel

File: Gemfile (hunk 1 above)
Comment (lomeroe) [reply Gemfile]: oops, sure

File: lib/kitchen/provisioner/salt_solo.rb (hunk 2 above)
Comment (lomeroe) [reply lib/kitchen/provisioner/salt_solo.rb]: I can remove it so it is like the non-windows side, the root_path directory was originally recursively cleaned out for non-windows, but I did notice the change on that one while rebasing (I honestly don't have any thoughts either way if it needs to be done or not)

Comment (gtmanfred) [reply lib/kitchen/provisioner/salt_solo.rb]: I would say for now lets make them the same.

**Review by gtmanfred — APPROVED** [review]

**Comment by gtmanfred** [comment]
alright, @barkingfoodog you are up, if you can get the windows tests in, I can cut a new release of kitchen-salt.

Thanks!
Daniel

**Comment by lomeroe** [comment]
My config still works w/#72 merged, but I specify pillars in the .kitchen.yml and have is_file_roots set to true.  I don't *see* anything that makes me think it wouldn't work, but I will admit my brain is a little mushy right now and trying to figure out what all is going on there/intended, so I could test other scenarios, is not going well for me...

**Comment by gtmanfred** [comment]
Yeah, no worries, I am going to take care of it once we have tests for windows.

**Comment by gtmanfred** [comment]
Thanks for the help!

**Comment by barkingfoodog** [comment]
@gtmanfred will do!

**Pull request merged**